Add Genebass parser

ireneisdoomed opened the pull request:
This PR includes:
- A new module to parse the Genebass burden results
- The integration of this module in GeneBurden.py
- Updates in the logic that imports the trait mappings for Genebass and AZ with a common util
- Integration in the pipeline: the snakemake rule runs without problems

Genebass is adding 8808 evidence making a total of 34485 for the whole burden evidence set.
This is an example evidence string:
```
{
    "allelicRequirements": null,
    "ancestry": "EUR",
    "ancestryId": "HANCESTRO_0009",
    "beta": -0.0036585665,
    "betaConfidenceIntervalLower": -0.0043895182,
    "betaConfidenceIntervalUpper": -0.0029276148,
    "cohortId": "UK Biobank 450k",
    "datasourceId": "gene_burden",
    "datatypeId": "genetic_association",
    "diseaseFromSource": "Whole body fat mass",
    "diseaseFromSourceId": "23100",
    "diseaseFromSourceMappedId": "EFO_0005409",
    "literature": null,
    "oddsRatio": null,
    "oddsRatioConfidenceIntervalLower": null,
    "oddsRatioConfidenceIntervalUpper": null,
    "pValueExponent": -7,
    "pValueMantissa": 5.58,
    "projectId": "Genebass",
    "resourceScore": 5.58e-7,
    "statisticalMethod": "missense|LC",
    "statisticalMethodOverview": "Burden test carried out with rare missense variants including low-confidence pLOF and in-frame indels.",
    "studyCases": 387351,
    "studyCasesWithQualifyingVariants": null,
    "studyId": null,
    "studySampleSize": 387351,
    "targetFromSourceId": "ENSG00000165572"
}
```

The evidence file is uploaded in its bucket (`gs://otar000-evidence_input/GeneBurden/json/gene_burden-2022-06-16.json.gz`) and is valid against the `il-burden-literature` branch of the JSON schema.

DSuveges commented on modules/GenebassGeneBurden.py:
```diff
@@ -0,0 +1,238 @@
+        raise AssertionError(
+            f"There are {evd_df.filter(col('resourceScore') == 0).count()} evidence with a P value of 0."
+        )
+    if not 8_000 < evd_df.count() < 9_000:
```
Can we assume this resource getting bigger? If so the upper boundary might be too stringent.

ireneisdoomed replied:
We can. As you say, this is an expectation I _currently_ have but which might change. Basically I prefer to pick up an increase when generating the evidence than when we later check the metrics. However I agree that runtime is not the place to do this quality checks

ireneisdoomed commented on modules/GenebassGeneBurden.py:
```diff
@@ -0,0 +1,238 @@
+    logging.warning(
+        f"There are {genebass_df.filter(col('Pvalue_Burden') == 0.0).count()} evidence with a p-value of 0.0."
+    )
+    minimum_pvalue = (
+        genebass_df.filter(col('Pvalue_Burden') > 0.0).agg({'Pvalue_Burden': 'min'}).collect()[0]['min(Pvalue_Burden)']
+    )
+    genebass_df = genebass_df.withColumn(
+        'Pvalue_Burden', when(col('Pvalue_Burden') == 0.0, lit(minimum_pvalue)).otherwise(col('Pvalue_Burden'))
+    )
```
Can you remind me what is the effect of a very low p Value? Was it that Spark was casting this to `Infinity` or something else? If the concern is the casting, we would pick this at validation.

ireneisdoomed commented on configuration.yaml:
```diff
@@ -26,9 +26,9 @@ EPMC:
   curation: https://raw.githubusercontent.com/opentargets/curation/master/gene_burden/curated_evidence.tsv
-  inputBucket: gs://otar000-evidence_input/GeneBurden/data_files
+  genebass: gs://otar000-evidence_input/GeneBurden/data_files/genebass_entries
+  traitMappings: https://github.com/opentargets/curation/blob/master/mappings/disease/manual_string.tsv?raw=true
```
Thanks! This is much better because in this way I can read the remote file with SparkFiles just fine. The general approach is something like:
```python
spark.sparkcontext.addFile(url)

spark.read.csv(SparkFiles.get('data.csv'))
```

Having the `raw` parameter at the tail of the url made the thing just cumbersome.

ireneisdoomed commented on modules/GenebassGeneBurden.py:
```diff
@@ -0,0 +1,238 @@
+    logging.info(f'File with the Genebass traits with their EFO mappings: {genebass_trait_mappings}')
+
+    # Load data
+    genebass_trait_mappings_df = read_trait_mappings(
```
[8c3308f](https://github.com/opentargets/evidence_datasource_parsers/pull/130/commits/8c3308f0a5fdd6b13316ef46348ee00ba6f7c650) contains changes about this. I simplified the function and hardcoded the URL of the table, so that the parsers need minimum logic to import it.

ireneisdoomed commented:
Thank you for all your comments @DSuveges. I've addressed them all and the current code is working, the evidence file has also been updated and uploaded.

From what I read, Spark's function to get the current session is `getActiveSession`. I've updated the `import_trait_mappings` function to use it so that we don't have to pass the spark session.
Thank you for the suggestion, I'll apply this approach in other parts of the code in a specific PR next week!

DSuveges commented:
Great, thanks a lot!

DSuveges reviewed (approved):
All makes sense
